openflow13/nx_action.go

```diff
@@ -0,0 +1,956 @@
+	binary.BigEndian.PutUint32(data[n:], fieldHeaderData)
+	n += 4
+	binary.BigEndian.PutUint64(data[n:], a.Value)
+	n += 8
+
+	return
+}
+
+func (a *NXActionRegLoad) UnmarshalBinary(data []byte) error {
+	n := 0
+	a.NXActionHeader = new(NXActionHeader)
+	err := a.NXActionHeader.UnmarshalBinary(data[n:])
+	n += int(a.NXActionHeader.Len())
+	if len(data) < int(a.Len()) {
+		return errors.New("the byte array has wrong size to unmarshal an NXActionRegLoad message")
+	}
+	a.OfsNbits = binary.BigEndian.Uint16(data[n:])
+	n += 2
+	a.DstReg = new(MatchField)
+	err = a.DstReg.UnmarshalHeader(data[n : n+4])
+	n += 4
+	a.Value = binary.BigEndian.Uint64(data[n:])
+	return err
+}
+
+// nx_action_reg_move
+type NXActionRegMove struct {
+	*NXActionHeader
+	Nbits    uint16
+	SrcOfs   uint16
+	DstOfs   uint16
+	SrcField *MatchField
+	DstField *MatchField
+}
+
+func NewNXActionRegMove(nBits uint16, srcOfs uint16, dstOfs uint16, srcField *MatchField, dstField *MatchField) *NXActionRegMove {
+	a := new(NXActionRegMove)
+	a.NXActionHeader = NewNxActionHeader(NXAST_REG_MOVE)
+	a.Length = a.NXActionHeader.Len() + 14
+	a.Nbits = nBits
+	a.SrcOfs = srcOfs
+	a.DstOfs = dstOfs
+	a.SrcField = srcField
+	a.DstField = dstField
+	return a
+}
+
+func (a *NXActionRegMove) Len() (n uint16) {
+	return a.Length
+}
+
+func (a *NXActionRegMove) MarshalBinary() (data []byte, err error) {
+	data = make([]byte, 24)
+	b := make([]byte, 0)
+	n := 0
+
+	b, err = a.NXActionHeader.MarshalBinary()
+	copy(data[n:], b)
+	n += len(b)
+	binary.BigEndian.PutUint16(data[n:], a.Nbits)
+	n += 2
+	binary.BigEndian.PutUint16(data[n:], a.SrcOfs)
+	n += 2
+	binary.BigEndian.PutUint16(data[n:], a.DstOfs)
+	n += 2
+
+	srcFieldHeaderData := a.SrcField.MarshalHeader()
+	binary.BigEndian.PutUint32(data[n:], srcFieldHeaderData)
+	n += 4
+
+	dstFieldHeaderData := a.DstField.MarshalHeader()
+	binary.BigEndian.PutUint32(data[n:], dstFieldHeaderData)
+	n += 4
+	return
+}
+
+func (a *NXActionRegMove) UnmarshalBinary(data []byte) error {
+	n := 0
+	a.NXActionHeader = new(NXActionHeader)
+	err := a.NXActionHeader.UnmarshalBinary(data[n:])
+	n += int(a.NXActionHeader.Len())
+	if len(data) < int(a.Len()) {
+		return errors.New("the byte array has wrong size to unmarshal an NXActionRegLoad message")
+	}
+	a.Nbits = binary.BigEndian.Uint16(data[n:])
+	n += 2
+	a.SrcOfs = binary.BigEndian.Uint16(data[n:])
+	n += 2
+	a.DstOfs = binary.BigEndian.Uint16(data[n:])
+	n += 2
+	a.SrcField = new(MatchField)
+	err = a.SrcField.UnmarshalHeader(data[n:])
+	n += 4
+	a.DstField = new(MatchField)
+	err = a.DstField.UnmarshalHeader(data[n:])
+	return err
+}
+
+type NXActionResubmit struct {
+	*NXActionHeader
+	InPort  uint16
+	TableID uint8
+	pad     []byte // 3 bytes
+}
+
+func NewNXActionResubmit(inPort uint16) *NXActionResubmit {
+	a := new(NXActionResubmit)
+	a.NXActionHeader = NewNxActionHeader(NXAST_RESUBMIT)
+	a.Type = Type_Experimenter
+	a.Length = a.NXActionHeader.Len() + 6
+	a.InPort = inPort
+	a.pad = make([]byte, 3, 3)
```

Comment:
There is no point in providing a capacity value if it is equal to the size (couple places below as well).

Reply:
Done.